build/flaws.js

```diff
@@ -497,10 +500,33 @@ async function fixFixableFlaws(doc, options, document) {
           timeout: 10000,
           retry: 3,
         });
+        const fileType = await FileType.fromBuffer(imageBuffer);
+        if (!fileType) {
+          throw new Error(
+            `No file type could be extracted from ${flaw.src} at all. Probably not going to be a valid image file.`
+          );
+        }
+        const isSVG =
+          fileType.mime === "application/xml" &&
+          flaw.src.toLowerCase().endsWith(".svg");
+
+        if (!(VALID_MIME_TYPES.has(fileType.mime) || isSVG)) {
+          throw new Error(
+            `${flaw.src} has an unrecognized mime type: ${fileType.mime}`
+          );
+        }
+        // Otherwise FileType would make it `.xml`
+        const imageExtension = isSVG ? "svg" : fileType.ext;
+        const imageBasename = sanitizeFilename(
+          `${path.basename(
+            decodeURI(url.pathname).replace(/\s+/g, "_"),
+            path.extname(decodeURI(url.pathname))
+          )}.${imageExtension}`
+        );
```

Comment:
```suggestion
        const decodedPathname = decodeURI(url.pathname).replace(/\s+/g, "_");
        const imageBasename = sanitizeFilename(
          `${path.basename(decodedPathname, path.extname(decodedPathname))
          }.${imageExtension}`
        );
```

In the unlikely case of a file called `foo.J PEG`